tools/lambda-promtail/lambda-promtail/promtail_client.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/dskit/backoff"
+)
+
+type IPromtailClient interface {
```

Comment:
we can just name the interface `client`

Reply:
Sure, will do! 👍  Actually I thought that it might get confusing later because there are other clients (s3clients etc...) that could be mocked and passed around using interfaces.  I guess it's not an issue to call this interface `client` for now and we still can move the clients to dedicated packages later on.